packages/react-core/src/components/Drawer/Drawer.tsx

```diff
@@ -31,23 +30,21 @@ export const DrawerContext = React.createContext<Partial<DrawerContextProps>>({
   isStatic: false
 });
 
+const timeout = Number.parseInt(c_drawer__panel_TransitionDuration.value.match(/\d+/)[0]);
+
 export const Drawer: React.SFC<DrawerProps> = ({
   className = '',
   children,
   isExpanded = false,
   isInline = false,
   isStatic = false,
   position = 'right',
-  onMount = () => {},
-  onUnmount = () => {},
+  onExpand = () => {},
   ...props
 }: DrawerProps) => {
   React.useEffect(() => {
-    onMount();
-    return () => {
-      onUnmount();
-    };
-  }, [isExpanded, onMount, onUnmount]);
+    setTimeout(onExpand, timeout);
```

Comment:
This isn't really a lifecycle function. It could read
```js
if (isExpanded) {
  setTimeout(onExpand, timeout);
}
```

I'll update it to this to avoid confusion.

Reply:
Previously the user had the ability to pass a custom cleanup function to `onUnmount`. is there a reason you didn't want to preserve it in some way? Or to clean up the timeout with clearTimeout? In case of an unnecessary re-render could we then have multiple timeouts going and completing in short succession?

Reply:
Sry just confused, I thought it would be necessary to clean up the timeout but I find the [docs on effects](https://reactjs.org/docs/hooks-effect.html) a bit unclear so just trying to understand better.

Reply:
Previously `onUnmount` would fire every single time `onMount` would fire. It's default of `() => {}` was not cleaning up anything.

Reply:
Right, `onUnmount` would fire and then `onMount` fires. In case a user has started a process-intensive and cancellable operation upon opening the drawer, they'd have the ability to add code to cancel it or do whatever cleanup may be necessary if when drawer is closed/re-opened.

Regardless of when that code runs, I'm still not clear on why it's being removed. Is that API part problematic?

I get if it makes sense to rename as `onExpand` and `onUnexpand`/`onCollapse` for clarity, but I don't understand why we would remove the ability to provide generic teardown logic altogether.

Reply:
It doesn't make sense to support a non-lifecycle callback to a user function. They can add their own callback to the end of their own function or chain promises however they want.

The previous `onUnmount` feature just simply doesn't make sense. The old code was basically "when the properties `isExpanded`, `onMount`, or `onUnmount` change, run `onMount()` then `onUnmount()`."